### PR TITLE
feat: Add vulnerability section to Markdown formatter (Issue #46)

### LINKED ISSUE #46: Subtask 6/8: Add vulnerability section to Markdown formatter (Issue #2)
## Parent Issue

Part of #2: Feature Request: Add --check-cve option for Markdown output using OSV API

## Description

Update the Markdown formatter to include a vulnerability report section when vulnerability data is present in the SBOM response.

## Scope

This subtask updates the Markdown formatter adapter to display vulnerability information.

**Estimated changes**: ~120 lines (small-medium task)

## Dependencies

- **Depends on**: #41 (Subtask 1/8) - Domain models
- **Depends on**: #44 (Subtask 4/8) - DTO updates

### File to Update

`src/adapters/outbound/formatters/markdown_formatter.rs`

### Vulnerability Section Format

The new section should be appended after the existing SBOM content:

```markdown
## Vulnerability Report

**⚠️ Security Issues Detected**

The following packages have known security vulnerabilities:

| Package | Current Version | Fixed Version | CVSS | Severity | CVE ID |
|---------|----------------|---------------|------|----------|--------|
| pydantic | 2.12.5 | 2.12.6 | 9.8 | 🔴 CRITICAL | CVE-2024-1234 |
| requests | 2.31.0 | 2.32.2 | 7.5 | 🟠 HIGH | CVE-2024-5678 |
| jinja2 | 3.1.2 | 3.1.3 | 5.3 | 🟡 MEDIUM | CVE-2024-9012 |

---

*Vulnerability data provided by [OSV](https://osv.dev) under CC-BY 4.0*
```

### Implementation Approach

```rust
impl SbomFormatter for MarkdownFormatter {
    fn format_with_dependencies(
        &self,
        dependency_graph: &DependencyGraph,
        packages: Vec<EnrichedPackage>,
        metadata: &SbomMetadata,
    ) -> Result<String> {
        let mut output = String::new();
        
        // ... existing SBOM sections ...
        
        // NEW: Add vulnerability section if present
        if let Some(ref vuln_report) = metadata.vulnerability_report {
            output.push_str(&self.format_vulnerability_section(vuln_report));
        }
        
        Ok(output)
    }

    fn format(
        &self,
        packages: Vec<EnrichedPackage>,
        metadata: &SbomMetadata,
    ) -> Result<String> {
        let mut output = String::new();
        
        // ... existing SBOM sections ...
        
        // NEW: Add vulnerability section if present
        if let Some(ref vuln_report) = metadata.vulnerability_report {
            output.push_str(&self.format_vulnerability_section(vuln_report));
        }
        
        Ok(output)
    }
}

impl MarkdownFormatter {
    /// Formats vulnerability report section
    ///
    /// Returns markdown formatted vulnerability table with:
    /// - Package name and current version
    /// - Fixed version (if available)
    /// - CVSS score
    /// - Severity with emoji indicator
    /// - CVE identifier
    fn format_vulnerability_section(
        &self,
        vulnerabilities: &[PackageVulnerabilities],
    ) -> String {
        if vulnerabilities.is_empty() {
            return self.format_no_vulnerabilities();
        }

        let mut output = String::new();
        output.push_str("\n## Vulnerability Report\n\n");
        output.push_str("**⚠️ Security Issues Detected**\n\n");
        output.push_str("The following packages have known security vulnerabilities:\n\n");
        
        // Table header
        output.push_str("| Package | Current Version | Fixed Version | CVSS | Severity | CVE ID |\n");
        output.push_str("|---------|----------------|---------------|------|----------|--------|\n");
        
        // Table rows - one row per vulnerability
        for pkg_vuln in vulnerabilities {
            for vuln in pkg_vuln.vulnerabilities() {
                output.push_str(&format!(
                    "| {} | {} | {} | {} | {} {} | {} |\n",
                    pkg_vuln.package_name(),
                    pkg_vuln.current_version(),
                    vuln.fixed_version().unwrap_or("N/A"),
                    vuln.cvss_score().map_or("N/A".to_string(), |s| format!("{:.1}", s)),
                    vuln.severity().emoji(),
                    format!("{:?}", vuln.severity()).to_uppercase(),
                    vuln.id(),
                ));
            }
        }
        
        // Attribution
        output.push_str("\n---\n\n");
        output.push_str("*Vulnerability data provided by [OSV](https://osv.dev) under CC-BY 4.0*\n");
        
        output
    }

    /// Formats message when no vulnerabilities are found
    fn format_no_vulnerabilities(&self) -> String {
        let mut output = String::new();
        output.push_str("\n## Vulnerability Report\n\n");
        output.push_str("**✅ No Known Vulnerabilities**\n\n");
        output.push_str("No security vulnerabilities were found in the scanned packages.\n\n");
        output.push_str("---\n\n");
        output.push_str("*Vulnerability data provided by [OSV](https://osv.dev) under CC-BY 4.0*\n");
        output
    }
}
```

### Design Decisions

**Why separate rows for each vulnerability?**
- A package can have multiple CVEs
- Each CVE may have different CVSS scores and fixed versions
- Clearer for users to see all issues

**Why include "No vulnerabilities" message?**
- Confirms that check was performed (not just skipped)
- Positive security signal
- Distinguishes from "check not performed"

**Attribution placement:**
- At the end of vulnerability section
- Links to OSV website
- Complies with CC-BY 4.0 requirements

## Files to Update

1. `src/adapters/outbound/formatters/markdown_formatter.rs` - Add vulnerability section formatting

## Acceptance Criteria

- [ ] `format_vulnerability_section()` method added
- [ ] `format_no_vulnerabilities()` method added
- [ ] Vulnerability table includes all required columns
- [ ] Severity emojis displayed correctly (🔴🟠🟡🟢)
- [ ] CVSS score formatted to 1 decimal place
- [ ] "N/A" shown for missing fixed versions or CVSS scores
- [ ] Multiple vulnerabilities per package handled correctly
- [ ] Empty vulnerability list shows "No Known Vulnerabilities" message
- [ ] OSV attribution included in all cases
- [ ] Attribution links to https://osv.dev
- [ ] Attribution mentions CC-BY 4.0 license
- [ ] Section only added when `vulnerability_report` is `Some`
- [ ] Unit tests for vulnerability formatting:
  - [ ] Test with multiple vulnerabilities
  - [ ] Test with single vulnerability
  - [ ] Test with empty vulnerability list
  - [ ] Test with missing CVSS scores
  - [ ] Test with missing fixed versions
- [ ] `cargo fmt --all` passes
- [ ] `cargo clippy --all-targets --all-features -- -D warnings` passes
- [ ] `cargo test` passes

## Testing Strategy

**Unit tests**:
- Test vulnerability table generation with mock data
- Test edge cases (missing CVSS, missing fixed version)
- Test empty vulnerability list
- Verify markdown formatting is correct

**Example test data**:
```rust
let vuln = Vulnerability::new(
    "CVE-2024-1234".to_string(),
    Some(9.8),
    Severity::Critical,
    Some("2.12.6".to_string()),
    Some("SQL injection vulnerability".to_string()),
).unwrap();
```

## Notes

- Only Markdown format gets vulnerability section (not JSON)
- JSON format already includes all data - formatters handle it differently
- Follow existing Markdown formatting style
- Use emojis from `Severity::emoji()` method (defined in domain layer)
- Attribution text must be exact to comply with license

## Estimated Size

~120 lines (formatting methods + tests)

### PR DESCRIPTION
**Pull request by Taketo-Yoda**
## Summary
- Updated Markdown formatter to display vulnerability information from CVE checks
- Extended SbomFormatter trait to pass vulnerability_report to all formatters
- Added comprehensive tests for vulnerability formatting

### Trait Changes
- Extended `SbomFormatter` trait methods to accept `vulnerability_report: Option<&[PackageVulnerabilities]>` parameter
- Updated both `format()` and `format_with_dependencies()` methods

### Markdown Formatter
- Added `format_vulnerability_section()` method to display vulnerabilities in table format
- Added `format_no_vulnerabilities()` method for clean checks
- Vulnerability table includes:
  - Package name and current version
  - Fixed version (if available)
  - CVSS score (formatted to 1 decimal place)
  - Severity with emoji indicators (🔴 Critical, 🟠 High, 🟡 Medium, 🟢 Low)
  - CVE identifier
- OSV attribution for CC-BY 4.0 compliance

### Other Formatters
- Updated `CycloneDxFormatter` to accept new parameter (ignored for JSON format)

### Main Application
- Updated `main.rs` to pass vulnerability_report from SbomResponse to formatters

## Test Coverage
- ✅ Test with multiple vulnerabilities
- ✅ Test with single vulnerability  
- ✅ Test with empty vulnerability list (no vulnerabilities found)
- ✅ Test without vulnerability check (None)
- ✅ Test with dependencies and vulnerabilities
- ✅ Test with missing CVSS scores
- ✅ Test with missing fixed versions
- ✅ All existing tests updated and passing

## Validation
- ✅ `cargo fmt --all` passes
- ✅ `cargo clippy --all-targets --all-features -- -D warnings` passes
- ✅ `cargo test` passes (177 tests)

## Related Issues
- Part of #2: Feature Request: Add --check-cve option
- Closes #46

## Screenshots
N/A (terminal output feature)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/adapters/outbound/formatters/cyclonedx_formatter.rs [hunk 1]
```diff
@@ -72,7 +73,12 @@ impl Default for CycloneDxFormatter {
 }
 
 impl SbomFormatter for CycloneDxFormatter {
-    fn format(&self, packages: Vec<EnrichedPackage>, metadata: &SbomMetadata) -> Result<String> {
+    fn format(
+        &self,
+        packages: Vec<EnrichedPackage>,
+        metadata: &SbomMetadata,
+        _vulnerability_report: Option<&[PackageVulnerabilities]>,
```
Comment (Taketo-Yoda) [inline src/adapters/outbound/formatters/cyclonedx_formatter.rs]: Although _vulnerability_report is prefixed with an underscore, please document the reason for its non-use within the format method's logic to improve maintainability for future developers.
